### PR TITLE
STACK-968 and STACK-973

### PR DESCRIPTION
**Pull request by omkar-bhujbal11**
Added Test Folders which contain UT for all 3 modules , also added travis.yml and passed pep8 ,ansible-test Testcases

### CODE REVIEW

File: galaxy.yml [hunk 1]
```diff
@@ -1,42 +1,14 @@
...
-# A list of the collection's content authors. Can be just the name or in the format 'Full Name <email> (url)
-# @nicks:irc/im.site#channel'
 authors:
-- your name <example@domain.com>
-
-
-### OPTIONAL but strongly recommended
-
-# A short summary description of the collection
+    - "omkar-bhujbal11"
```
Comment (hthompson6) [inline galaxy.yml]: Please change this to opensource@a10networks.com

File: galaxy.yml [hunk 2]
```diff
@@ -45,7 +17,7 @@ tags: []
 dependencies: {}
 
 # The URL of the originating SCM repository
-repository: http://example.com/repository
+repository: https://github.com/omkar-bhujbal11/a10-cliconf-collection/tree/STACK-968
```
Comment (hthompson6) [inline galaxy.yml]: Please change this to https://github.com/a10networks/a10-cliconf-collection

File: README.md [hunk 3]
```diff
@@ -1,3 +1,79 @@
...
+		ANSIBLE_COLLECTIONS_PATHS=<path-to-collections-folders>
+	
+	3) Add below line in /etc/ansible/ansible.cfg File 
+		
+		collections_paths=<path-to-collection1>:<path-to-collection2>
+
+	4) Keep your playbooks to run in relative to collection 
+
+			|── myplaybook.yml
+			├── collections/
```
Comment (hthompson6) [inline README.md:50]: Why does this have so many directories?

File: plugins/modules/acos_config.py [hunk 4]
```diff
@@ -350,7 +343,7 @@ def main():
                 if not line.startswith('!'):
                     run_commands(module, line.strip())
             run_commands(module, 'exit')
-        except IOError as e:
+        except IOError:
             module.fail_json(msg="File Not Found!")
```
Comment (hthompson6) [inline plugins/modules/acos_config.py]: Let's include file path + name on this fail_json string.

File: tests/unit/modules/network/a10/test_acos_facts.py [hunk 5]
```diff
@@ -0,0 +1,142 @@
...
+        set_module_args(dict(gather_subset='interfaces'))
+        result = self.execute_module()
+        self.assertEqual(
+            result['ansible_facts']['ansible_net_interfaces'][
+                'Ethernet 1']['mtu'], 1500
+        )
+        self.assertEqual(
+            result['ansible_facts']['ansible_net_interfaces'][
+                'Ethernet 2']['ipv6'][0]['address'],
+            '2001:db8:85a3::8a2e:370:7334'
```
Comment (hthompson6) [inline tests/unit/modules/network/a10/test_acos_facts.py]: Indentation

File: tests/unit/modules/network/a10/test_acos_facts.py [hunk 6]
```diff
@@ -0,0 +1,142 @@
...
+        )
+        self.assertEqual(
+            result['ansible_facts']['ansible_net_interfaces'][
+                'Ethernet 3']['macaddress'],
+            'fa16.3e6d.95ec'
```
Comment (hthompson6) [inline tests/unit/modules/network/a10/test_acos_facts.py]: Indentation

File: plugins/modules/acos_config.py [hunk 7]
```diff
@@ -294,8 +296,6 @@ def main():
         before=dict(type='list'),
         after=dict(type='list'),
 
-        match=dict(default='line', choices=['line', 'strict',
```
Comment (hthompson6) [inline plugins/modules/acos_config.py:297]: Why was this removed from argspec?

**Review by hthompson6 — CHANGES_REQUESTED** [review]

**Comment by hthompson6** [comment]
Copying code from ansible and pasting it to our modules isn't a viable work around.

https://github.com/ansible-collections/ansible.netcommon

ansible-collections should be shipped with ansible and made available to us.

**Comment by hthompson6** [comment]
tests/unit/modules/network/a10/fixtures/acos_facts_show_hardware_section_Hardware <- empty file. Why include it in test fixtures?

**Comment by hthompson6** [comment]
A lot of the fixtures included are near copy-paste to one another. Is it possible to reuse them?

  Also for the oneline fixtures. Why not just inject the one string you want to test against?

**Comment by hthompson6** [comment]
https://github.com/a10networks/a10-cliconf-collection/pull/2/files#diff-9e9e581e045cd219b0d6deb7998d88b9R391  

Can you please change that lowercase C in "check" to "Check." Not part of the review but would be nice to fix now along with some of the other refactoring that was done.

**Comment by hthompson6** [comment]
There's only two commits in this PR. And they don't tell me too much about what salient changes occured in the code. For 66 modules, I'd expect to see a far longer commit history.

File: README.md (hunk 3 above)
Comment (omkar-bhujbal11) [reply README.md:50]: This is just an example that we have used from Ansible collection documentation. It has so many directories because it structures of ansible collection which we get after installing it

File: plugins/modules/acos_config.py (hunk 4 above)
Comment (omkar-bhujbal11) [reply plugins/modules/acos_config.py]: We added file path which already includes the  filename in it

File: plugins/modules/acos_config.py (hunk 7 above)
Comment (omkar-bhujbal11) [reply plugins/modules/acos_config.py:297]: we refactored acos_config.py for the first time and we do not support this match argument for acos_config

**Comment by omkar-bhujbal11** [comment]
> Copying code from ansible and pasting it to our modules isn't a viable work around.
> 
> https://github.com/ansible-collections/ansible.netcommon
> 
> ansible-collections should be shipped with ansible and made available to us.

we have made changes accordingly

**Comment by omkar-bhujbal11** [comment]
> tests/unit/modules/network/a10/fixtures/acos_facts_show_hardware_section_Hardware <- empty file. Why include it in test fixtures?

Modified that fixture file

**Comment by omkar-bhujbal11** [comment]
> A lot of the fixtures included are near copy-paste to one another. Is it possible to reuse them?
> 
>   Also for the oneline fixtures. Why not just inject the one string you want to test against?

We have removed the unused fixtures file. We are not injecting one line because we are using the results that we are getting from the ACOS device in that fixture file. we don't want them to hardcode so we are loading them from the respective fixture file using load_fixtures function

**Comment by omkar-bhujbal11** [comment]
> There's only two commits in this PR. And they don't tell me too much about what salient changes occured in the code. For 66 modules, I'd expect to see a far longer commit history.

There were many commits because we were testing for pep8 and ansible-test using Travis . So we squashed that git commits.  In the future, we will keep all the commits.

**Review by hthompson6 — APPROVED** [review]
